Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/CategorySample.cs

```diff
@@ -0,0 +1,258 @@
+
+                    if (perfObject.ObjectNameTitleIndex == categoryNameIndex)
+                    {
+                        foundCategory = true;
+                        break;
+                    }
+
+                    dataRef = (IntPtr)((long)dataRef + perfObject.TotalByteLength);
+                }
+
+                if (!foundCategory)
+                {
+                    throw new InvalidOperationException("Category not found for categoryNameIndex " + categoryNameIndex);
+                }
+
+                int counterNumber = perfObject.NumCounters;
+                int instanceNumber = perfObject.NumInstances;
+
+                bool isMultiInstance = instanceNumber != -1;
+
+                // Move pointer forward to end of PERF_OBJECT_TYPE
+                dataRef = (IntPtr)((long)dataRef + perfObject.HeaderLength);
+
+                CounterDefinitionSample[] samples = new CounterDefinitionSample[counterNumber];
+                CounterDefinitionSample sample = null;
+                this.CounterTable = new Dictionary<int, CounterDefinitionSample>(counterNumber);
+                for (int index = 0; index < samples.Length; ++index)
+                {
+                    NativeMethods.PERF_COUNTER_DEFINITION perfCounter = new NativeMethods.PERF_COUNTER_DEFINITION();
+                    Marshal.PtrToStructure(dataRef, perfCounter);
+
+                    samples[index] = new CounterDefinitionSample(perfCounter, instanceNumber);
+                    if (perfCounter.CounterNameTitleIndex == counterNameIndex)
+                    {
+                        sample = samples[index];
+                    }
+
+                    dataRef = (IntPtr)((long)dataRef + perfCounter.ByteLength);
+
+                    int currentSampleType = samples[index].CounterType;
+                    if (!CategorySample.IsBaseCounter(currentSampleType))
+                    {
+                        // We'll put only non-base counters in the table. 
+                        if (currentSampleType != NativeMethods.PERF_COUNTER_NODATA)
+                        {
+                            this.CounterTable[samples[index].NameIndex] = samples[index];
+                        }
+                    }
+                    else
+                    {
+                        // ignore base counters
+                    }
+                }
+
+                if (sample == null)
+                {
+                    throw new InvalidOperationException("Could not find the counter " + counterNameIndex);
+                }
+
+                // now set up the InstanceNameTable.  
+                if (!isMultiInstance)
+                {
+                    throw new InvalidOperationException("Single instance categories are not supported");
+                }
+
+                string[] parentInstanceNames = null;
+                this.InstanceNameTable = new Dictionary<string, int>(instanceNumber, StringComparer.OrdinalIgnoreCase);
```

Comment:
Yes, I have tested it. With PIDs enabled, the names are not unique anymore, so we'll get the process_pid format, and won't add any #'s here.